youtube_dl/extractor/pornhub.py

```diff
@@ -518,9 +518,9 @@ def _extract_entries(self, webpage, host):
         return [
             self.url_result(
                 'http://www.%s/%s' % (host, video_url),
-                PornHubIE.ie_key(), video_title=title)
-            for video_url, title in orderedSet(re.findall(
-                r'href="/?(view_video\.php\?.*\bviewkey=[\da-z]+[^"]*)"[^>]*\s+title="([^"]+)"',
+                PornHubIE.ie_key(), video_title=title, video_id=id)
+            for video_url, id, title in orderedSet(re.findall(
+                r'href="/?(view_video\.php\?.*\bviewkey=([\da-z]+)[^"]*)"[^>]*\s+title="([^"]+)"',
```

Comment:
Actually, if extracting the ID, that's the primary key on which we should be filtering.
```py
        seen_ids = set()
        for m in re.finditer(
                r'href="/?(view_video\.php\?.*\bviewkey=([\da-z]+)[^"]*)"[^>]*\s+title="([^"]+)"',
                container):
            v_url, v_id, title = m.group(1, 2, 3)
            if v_id in seen_ids:
                continue
            yield self.url_result(
                'http://www.%s/%s' % (host, v_url),
                PornHubIE.ie_key(), video_title=title, video_id=v_id)
            seen_ids.add(v_id)
```
Anyhow, avoid a variable `id` since that has a specific meaning in Python.
```suggestion
                PornHubIE.ie_key(), video_title=title, video_id=v_id)
            for video_url, v_id, title in orderedSet(re.findall(
                r'href="/?(view_video\.php\?.*\bviewkey=([\da-z]+)[^"]*)"[^>]*\s+title="([^"]+)"',
```
Also, the question as to why `orderedSet()`, which is de-duplicated but not sorted, got its name arises ...